sovrin_node/server/domain_req_handler.py

```diff
@@ -0,0 +1,346 @@
+        nym = msg.get(TARGET_NYM)
+        if self.hasNym(nym, isCommitted=False):
+            if not self.idrCache.hasTrustee(identifier, isCommitted=False) and \
+                            self.idrCache.getOwnerFor(nym, isCommitted=False) != identifier:
+                reason = '{} is neither Trustee nor owner of {}'\
+                    .format(identifier, nym)
+                return False, reason
+        return True, ''
+
+    def doStaticValidation(self, identifier, reqId, operation):
+        if operation[TXN_TYPE] == NYM:
+            role = operation.get(ROLE)
+            nym = operation.get(TARGET_NYM)
+            if not nym:
+                raise InvalidClientRequest(identifier, reqId,
+                                           "{} needs to be present".
+                                           format(TARGET_NYM))
+            if not Authoriser.isValidRole(role):
+                raise InvalidClientRequest(identifier, reqId,
+                                           "{} not a valid role".
+                                           format(role))
+            s, reason = self.canNymRequestBeProcessed(identifier, operation)
+            if not s:
+                raise InvalidClientRequest(identifier, reqId, reason)
+
+        if operation[TXN_TYPE] == ATTRIB:
+            dataKeys = {RAW, ENC, HASH}.intersection(set(operation.keys()))
+            if len(dataKeys) != 1:
+                raise InvalidClientRequest(identifier, reqId,
+                                           '{} should have one and only one of '
+                                           '{}, {}, {}'
+                                           .format(ATTRIB, RAW, ENC, HASH))
+            if RAW in dataKeys:
+                try:
+                    data = json.loads(operation[RAW])
+                    endpoint = data.get(ENDPOINT, {}).get('ha')
+                    check_endpoint_valid(endpoint, required=False)
+
+                except EndpointException as exc:
+                    raise InvalidClientRequest(identifier, reqId, str(exc))
+                except BaseException as exc:
+                    raise InvalidClientRequest(identifier, reqId, str(exc))
+                    # PREVIOUS CODE, ASSUMED ANY EXCEPTION WAS A JSON ISSUE
+                    # except:
+                    #     raise InvalidClientRequest(identifier, reqId,
+                    #                                'raw attribute {} should be '
+                    #                                'JSON'.format(operation[RAW]))
+
+            if not (not operation.get(TARGET_NYM) or
+                    self.hasNym(operation[TARGET_NYM], isCommitted=False)):
+                raise InvalidClientRequest(identifier, reqId,
+                                           '{} should be added before adding '
+                                           'attribute for it'.
+                                           format(TARGET_NYM))
+
+    def validate(self, req: Request, config=None):
```

Comment:
Please split the method to perform validation for each txn type (like doValidateNym, doValidateAttr)